vendor/sigs.k8s.io/cluster-api/pkg/controller/machine/controller.go

```diff
@@ -107,7 +108,35 @@ func (c *MachineControllerImpl) Reconcile(machine *clusterv1.Machine) error {
 			glog.Infof("Skipping reconciling of machine object %v", name)
 			return nil
 		}
+
 		glog.Infof("reconciling machine object %v triggers delete.", name)
+		if machine.Labels["force-deletion"] != "True" && machine.Status.NodeRef != nil {
```

Comment:
```forceDeletion, exists := machine.Labels["force-deletion"]; !exists || (forceDeletion != "True" && machine.Status.NodeRef != nil)```

- if the `force-deletion` label is not set => drain the node
- if the `force-deletion` label is set, set to `True` and the `NodeRef` is set => drain the node

Reply:
Ah, I just meant to comment.